.spike-codex-appserver/schema/regen.sh

```
@@ -0,0 +1,37 @@
+echo "generating codex $VERSION protocol schema..."
+npx -y "@openai/codex@$VERSION" app-server generate-json-schema --out "$TMP"
+
+mkdir -p "$OUT"
+for f in ClientRequest.json ServerRequest.json ServerNotification.json ClientNotification.json; do
+  cp "$TMP/$f" "$OUT/$f"
+done
+node "$HERE/manifest.mjs" "$TMP" "$VERSION" > "$OUT/MANIFEST.json"
+echo "wrote $OUT ($(ls "$OUT" | wc -l) files)"
+
+if [ "${2:-}" = "--diff" ]; then
+  PREV="$(ls -d "$HERE"/*/ | grep -v "/$VERSION/$" | sort -V | tail -1)"
```

Shellcheck (0.11.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)